### PR TITLE
blockchain/v2: make the removal of an already removed peer a noop

### LINKED ISSUE #5541: blockchain/v2: tried to remove peer in peerStateRemoved
E2E tests report these errors, which caused the validator to take a very long time to fast sync.

https://github.com/tendermint/tendermint/runs/1284398417

```
validator03    | E[2020-10-21|02:13:35.702] Failed to write PacketMsg                    module=p2p peer=d3b5bfd2aeac159c79812dd9426de51a971c7a61@10.186.73.2:44898 err="write tcp 10.186.73.4:26656->10.186.73.2:44898: use of closed network connection"
validator03    | E[2020-10-21|02:13:35.702] Stopping peer for error                      module=p2p peer="Peer{MConn{10.186.73.2:44898} d3b5bfd2aeac159c79812dd9426de51a971c7a61 in}" err="write tcp 10.186.73.4:26656->10.186.73.2:44898: use of closed network connection"
validator03    | E[2020-10-21|02:13:35.702] error while stopping peer                    module=p2p error="already stopped"
validator03    | E[2020-10-21|02:13:35.702] Scheduler failure                            module=blockchain err="tried to remove peer d3b5bfd2aeac159c79812dd9426de51a971c7a61 in peerStateRemoved"
validator03    | E[2020-10-21|02:13:35.703] Scheduler failure                            module=blockchain err="tried to remove peer d3b5bfd2aeac159c79812dd9426de51a971c7a61 in peerStateRemoved"
validator03    | E[2020-10-21|02:13:35.812] Stopping peer for error                      module=p2p peer="Peer{MConn{10.186.73.2:44900} d3b5bfd2aeac159c79812dd9426de51a971c7a61 in}" err=scPeerError
```

### PR DESCRIPTION
**Pull request by melekes**
also, since multiple StopPeerForError calls may be executed in parallel,
only execute StopPeerForError once

Closes #5541

### CODE REVIEW

**Review by erikgrinaker — APPROVED** [review]
LGTM, remember a changelog entry

**Comment by melekes** [comment]
I[2020-10-22|16:29:08.969] Removing Docker containers and networks
I[2020-10-22|16:29:12.141] Removing testnet directory "networks/cif"
I[2020-10-22|16:29:12.142] Generating testnet files in "networks/cif"
I[2020-10-22|16:29:12.158] Starting initial network nodes...
I[2020-10-22|16:29:12.158] Starting transaction load (16 workers)...
I[2020-10-22|16:29:16.784] Node validator01 up on http://127.0.0.1:5701
I[2020-10-22|16:29:20.583] Node validator02 up on http://127.0.0.1:5702
I[2020-10-22|16:29:25.156] Node validator03 up on http://127.0.0.1:5703
I[2020-10-22|16:29:25.156] Waiting for initial height 1000...
I[2020-10-22|16:29:25.186] Starting node validator04 at height 1005...
I[2020-10-22|16:29:41.309] Node validator04 up on http://127.0.0.1:5704 at height 1005
I[2020-10-22|16:29:41.309] Disconnecting node validator04...
I[2020-10-22|16:29:52.887] Node validator04 recovered at height 1006
I[2020-10-22|16:29:55.945] Waiting for all nodes to reach height 1016...
I[2020-10-22|16:30:12.990] Ending transaction load after 154 txs (2.5 tx/s)...
I[2020-10-22|16:30:13.106] Waiting for all nodes to reach height 1021...
I[2020-10-22|16:30:21.237] Running tests in ./tests/...
ok      github.com/tendermint/tendermint/test/e2e/tests 5.185s

File: blockchain/v2/scheduler.go [hunk 1]
```diff
@@ -372,15 +365,9 @@ func (sc *scheduler) setStateAtHeight(height int64, state blockState) {
 	sc.blockStates[height] = state
 }
 
+// CONTRACT: peer exists and in Ready state.
```
Comment (erikgrinaker) [inline blockchain/v2/scheduler.go:368]: Should we really assume this? Concurrency is hard to reason about, isn't it optimistic to think we'll be able to enforce this?

Comment (melekes) [reply blockchain/v2/scheduler.go:368]: It isn't about concurrency. If you look at the code, you'll see that `touchPeer` func is called before this func. `touchPeer` ensures that the peer exists.

**Review by erikgrinaker — APPROVED** [review]

**Pull request merged**